### ISSUE TITLE
VS crash (and would not close - made Windows Server 2008 unresponsive but with low CPU)

### ISSUE THREAD

**Issue by flexiblefactory** [Microsoft/nodejstools#656]
Here is the stack trace:
```
Application: devenv.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.NullReferenceException
   at Microsoft.NodejsTools.Analysis.AnalysisUnit.AnalyzeWorker(Microsoft.NodejsTools.Analysis.Analyzer.DDG, System.Threading.CancellationToken)
   at Microsoft.NodejsTools.Analysis.Analyzer.DDG.Analyze(Microsoft.NodejsTools.Analysis.Deque`1<Microsoft.NodejsTools.Analysis.AnalysisUnit>, System.Threading.CancellationToken)
   at Microsoft.NodejsTools.Analysis.JsAnalyzer.AnalyzeQueuedEntries(System.Threading.CancellationToken)
   at Microsoft.NodejsTools.Intellisense.VsProjectAnalyzer+AnalysisQueue+GroupAnalysis.Analyze(System.Threading.CancellationToken)
   at Microsoft.NodejsTools.Intellisense.VsProjectAnalyzer+AnalysisQueue.Worker(System.Object)
   at System.Threading.ThreadHelper.ThreadStart_Context(System.Object)
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
   at System.Threading.ThreadHelper.ThreadStart(System.Object)
```

**Comment by flexiblefactory** [comment]
Third time this has happened in 24 hours. Almost completely locks up my machine when this happens. Things go into slow motion.... Eventually when my clicks register and the crash dialogue is dismissed, VS closes and things return to normal. But this takes more than 10 minutes.

**Comment by bartvanderwal** [comment]
NTVS also made my Visual Studio crash three times today. It's incredibly slow, and my hard drive starts swapping like a mad man. Never had this kind of extreme behaviour working on ASP.NET projects. I find it hard to believe that transpiling TS to JS would be harder than C# to IL. Could Resharper be causing this?

**Comment by liamprescott** [comment]
**Visual Studio**: 2015 update 1
**NTVS**: v.1.1

Visual Studio **continually** freezing and crashing ("Visual Studio has stopped working").

This appears to be due to a NTVS error as when VS process debugged in VS get the following:
```
NullReferenceException was unhandled
An unhandled exception of type 'System.NullReferenceException' occurred in Microsoft.NodejsTools.Analysis.dll

Additional information: Object reference not set to an instance of an object.
```
## These are the details from the Event Viewer > Event Properties

Log Name:      Application
Source:        Application Error
Date:          14/01/2016 09:25:53
Event ID:      1000
Task Category: (100)
Level:         Error
Keywords:      Classic
User:          N/A
Computer:      xxx-xxx-xxx
Description:
Faulting application name: devenv.exe, version: 14.0.24720.0, time stamp: 0x564ea97e
Faulting module name: unknown, version: 0.0.0.0, time stamp: 0x00000000
Exception code: 0xc0000005
Fault offset: 0x282a6a6c
Faulting process id: 0xf7c
Faulting application start time: 0x01d14ead1c2dab60
Faulting application path: C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\devenv.exe
Faulting module path: unknown
Report Id: ce2c1ae0-baa0-11e5-bcd5-9c45469ddbde
Event Xml:
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name="Application Error" />
    <EventID Qualifiers="0">1000</EventID>
    <Level>2</Level>
    <Task>100</Task>
    <Keywords>0x80000000000000</Keywords>
    <TimeCreated SystemTime="2016-01-14T09:25:53.000000000Z" />
    <EventRecordID>53021</EventRecordID>
    <Channel>Application</Channel>
    <Computer>xxx-xxx-xxx</Computer>
    <Security />
  </System>
  <EventData>
    <Data>devenv.exe</Data>
    <Data>14.0.24720.0</Data>
    <Data>564ea97e</Data>
    <Data>unknown</Data>
    <Data>0.0.0.0</Data>
    <Data>00000000</Data>
    <Data>c0000005</Data>
    <Data>282a6a6c</Data>
    <Data>f7c</Data>
    <Data>01d14ead1c2dab60</Data>
    <Data>C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\devenv.exe</Data>
    <Data>unknown</Data>
    <Data>ce2c1ae0-baa0-11e5-bcd5-9c45469ddbde</Data>
  </EventData>
</Event>

**This is making Visual Studio unusable.**

**Comment by mousetraps** [comment]
@flexiblefactory thanks for the stack trace. This looks like a duplicate of #70, which we've already fixed. Can you verify that the issue does not occur in the latest release? https://github.com/Microsoft/nodejstools

**Comment by mousetraps** [comment]
@bartvanderwal @liamprescott - would you mind grabbing the full exception details from the corresponding entry in Event Viewer? Also it would be helpful if you could also share your package.json for the project. Unfortunately we don't have enough information to diagnose just yet.

**Comment by nearwood** [comment]
package.json:
```
{
  "name": "TEST",
  "version": "0.0.0",
  "description": "",
  "main": "server/index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "start": "node server/index.js",
    "build-min-client-prod": "node client/www/build/private/buildScript.js",
    "build-min-client-dev": "node client/www/build/private/buildScript.js dev"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "async-waterfall": "^0.1.5",
    "babel-preset-es2015": "^6.6.0",
    "babel-preset-react": "^6.5.0",
    "babelify": "^7.2.0",
    "browserify": "^13.0.0",
    "browserify-css": "^0.9.1",
    "compression": "^1.6.1",
    "exorcist": "^0.4.0",
    "express": "^4.13.3",
    "fast-json-patch": "^0.5.6",
    "fs-extra": "^0.26.7",
    "jquery": "^1.12.2",
    "jsonpatch-to-mongodb": "^0.2.0",
    "mongodb": "^2.1.4",
    "ramda": "^0.20.0",
    "react": "^0.14.7",
    "react-addons-transition-group": "^0.14.7",
    "react-dom": "^0.14.7",
    "socket.io": "^1.4.4",
    "socket.io-client": "^1.4.5",
    "uglifyify": "^3.0.1",
    "watchify": "^3.7.0"
  },
  "devDependencies": {}
}

```

**Comment by mjbvz** [comment]
We've made some significant improvements to IntelliSense with NTVS 1.2 that should address many of the performance and crashes seen in the static analyzer.

The static analysis engine used by default in NTVS 1.1 can provide very powerful IntelliSense, but it also is our top source of crashes and has some major perf problem. So we decided to move to a new analysis engine powered by TypeScript. This provides more stable and consistent IntelliSense, and ultimately will allow us to support new JavaScript language features much more quickly.

I recommend downloading [NTVS 1.2 Beta](https://github.com/Microsoft/nodejstools/releases) and trying out the ES6 IntelliSense engine. We can still try to investigate specific crashes or hangs in NTVS 1.1, but will not be doing major feature work on the static analysis engine it uses.
